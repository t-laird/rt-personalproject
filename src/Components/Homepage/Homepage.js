import React from 'react';
import './Homepage.css';

const Homepage = () => {

  return (
    <div className="homepage-component">
      <div className="bg-holder">
			  <h1>WHERE STUDENTS <span className="no-wrap">ADV<img className='homepage-logo' alt="homepage-logo" src={require('./homepage-assets/logo.svg')} />CATE</span> STUDENTS</h1>
        <button className="header-button-top">
          <a href="https://tr-personal-proj.e1.loginrocket.com/">start SNAP NINJA now</a>
        </button>

        <div className="homepage-container">
          <div className="info-box">
            <img className='homepage-icon' alt="homepage-icon" src={require('./homepage-assets/handshake.svg')} />
            <h4>Encourage student-to-student mentorship</h4>
            <p>Receiving a little bit of recognition is powerful.  In Snap Ninja, students reward other students by giving each other snaps for helping them out.  Students receive weekly performance charts, providing a constant reminder of the importance of mentoring others.</p>
            <img className='homepage-graphic' alt="homepage-graphic" src={require('./homepage-assets/award-ss.svg')} />
          </div>

          <div className="info-box">
            <img className='homepage-icon' alt="homepage-icon" src={require('./homepage-assets/ribbon.svg')} />
            <h4>Reward collaboration</h4>
            <p>Rather than just walking away after working with others, Snap Ninja provides students an easy way to give helpful feedback.  Collaboration is both encouraged and rewarded, as it should be.</p>
            <img className='homepage-graphic' alt="homepage-graphic" src={require('./homepage-assets/welcome-ss.svg')} />
          </div>

          <div className="info-box">
            <img className='homepage-icon' alt="homepage-icon" src={require('./homepage-assets/mega.svg')} />
            <h4>Make shout-outs digital</h4>
            <p>Shoutouts are done the 2018 way, using easy-to-understand technology.  Thank yous are easily passed and long-term data is automatically tracked, giving students a personal record of the most important kind of achievement: Helping others succeed.</p>
            <img className='homepage-graphic' alt="homepage-graphic" src={require('./homepage-assets/notes-ss.svg')} />
          </div>

          <div className="info-box">
            <img className='homepage-icon' alt="homepage-icon" src={require('./homepage-assets/people.svg')} />
            <h4>Build intra-cohort relationships</h4>
            <p>Relationship-building is paramount.  Students are rewarded for getting to know others, without being punished for a slow day or week.  Group stats are tracked so students know theyâ€™re in it together.</p>
            <img className='homepage-graphic' alt="homepage-graphic" src={require('./homepage-assets/group-ss.svg')} />
          </div>
        </div>

        <button>
          <a href="https://tr-personal-proj.e1.loginrocket.com/">start SNAP NINJA now</a>
        </button>
      </div>
    </div>
  ); 
};

export default Homepage;