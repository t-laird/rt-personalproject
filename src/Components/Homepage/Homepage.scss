@import '../../Variables.scss';

.homepage-component {
  min-height: 100vh;
  background: url('homepage-assets/home-bg.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  background-position: 90% 50%;
  margin-top: -70px;
  background-attachment: fixed;

  .bg-holder {
    min-height: 100vh;
    padding-top: 158px;
    padding-bottom: 50px;
    background: linear-gradient(to bottom, rgba(50,50,50,.8) 0%,rgba(50,50,50,.5) 100%);
  }

  h1 {
    font-weight: 300;
    color: #fff;
    text-align: center;
    margin: 0 5px 142px;
    margin-bottom: 142px;
    line-height: 1.3;
  }

  h4 {
    font-size: 24px;
    font-weight: 900;
    color: $c3;
    line-height: 1.3;
    margin-bottom: 10px;
  }

  p {
    font-family: 'semplicitapro', sans-serif;
    font-weight: 500;
    color: $c3;
    font-size: 16px;
    opacity: .7;
    line-height: 1.3;
    margin-bottom: 10px;
  }

  .no-wrap {
    font-family: 'semplicitapro', 'sans-serif';
    font-size: 46px;
    font-weight: 300;
    white-space: nowrap;
  }

  .homepage-container {
    display: flex;
    flex-wrap: wrap;
    width: 1076px;
    margin: auto;
  }

  .info-box {
    background-color: $c5;
    border-left: 60px solid $c6;
    margin: 20px;
    padding: 24px 45px;
    position: relative;
    width: 498px;
    height: 437px;
  }

  .homepage-icon {
    width: 60px;
    position: absolute;
    top: 185px;
    left: -30px;
  }

  .homepage-logo {
    width: 50px;
    position: relative;
    top: 6px;
    left: 1px;
  }

  button {
    background-color: $c1;
    padding: 16px 23px 15px;
    border: 0;
    display: block;
    margin: 20px auto 0;
    cursor: pointer;
    transition: background-color .15s ease-in-out;

    a {
      color: #fff;
      }
  }

  button:hover {
    background-color: darken($c1, 12%);
  }

}

@media screen and (max-width: 1075px) {
  .homepage-component {
    .homepage-container {
      width: 538px;
    }
  }
}

@media screen and (max-width: 680px) {
  .homepage-component {
    h1, .no-wrap {
      font-size: 30px;
    }

    .homepage-logo {
      width: 36px;
    }
  }
}

@media screen and (max-width: 537px) {
  .homepage-component {

    h1 {
      margin-bottom: 110px;
    }

    .homepage-container {
      width: 100%;
    }

    .info-box {
      background-color: $c5;
      border-left: 0;
      border-top: 60px solid $c6;
      margin: 20px 0;
      width: 100%;
      height: unset;
    }

    .homepage-icon {
      width: 60px;
      position: static;
      display: block;
      margin: -54px auto 0;
    }
  }
}
