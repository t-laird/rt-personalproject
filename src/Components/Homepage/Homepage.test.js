import React from 'react';
import Homepage from './Homepage';
import { shallow } from 'enzyme';

describe('homepage tests', () => {
  it('should match the snapshot', () => {
    const renderedApp = shallow(<Homepage />);

    expect(renderedApp).toMatchSnapshot();
  })
})