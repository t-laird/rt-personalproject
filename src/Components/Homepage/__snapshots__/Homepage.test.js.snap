// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homepage tests should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Homepage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Homepage",
    },
    "ref": null,
    "rendered": "Homepage",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Homepage",
      },
      "ref": null,
      "rendered": "Homepage",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
